ObjectSpace.h

```diff
@@ -0,0 +1,395 @@
+//####################################################################
+//
+//    FILENAME:          ObjectSpace.h
+//
+//    CLASSIFICATION:    Unclassified
+//
+//    DESCRIPTION:
+//
+//    Header for object space definition class and non-ECEF ObjectSpace coordinate classes.
+//
+//    LIMITATIONS:       None
+//
+//
+//    SOFTWARE HISTORY:
+//     Date          Author    Comment
+//     -----------   ------    -------
+//     31-March-2022    Eugene Rose CCB Change CSM 3.0.4
+//
+//    NOTES:
+//
+//#####################################################################
+
+#ifndef __CSM_OBJECTSPACE_H
+#define __CSM_OBJECTSPACE_H
+
+namespace csm
+{
+enum class CSM_EXPORT_API ObjectSpaceType : unsigned char
+{
+    LSR,
+    ECI,
+    UNDEFINED
+};
+//>
+// This enum class identifies the object space type.
+//<
+    
+//***
+// STRUCT: ObjectSpaceCoordinate
+//> This structure represents a three-dimensional point in the object space coordinate system.
+//
+//  Coordinate values are in units of meters.
+//<
+//***
+struct CSM_EXPORT_API ObjectSpaceCoordinate
```

Comment:
I do like sminster's suggestion to simply

typedef EcefCoord ObjectSpaceCoordinate;

etc when possible. That would give SETs the option to interact with these new classes with existing code and/or same between ecef and close range vs writing separate code for interacting with close range vs ecf.